dmscripts/helpers/supplier_data_helpers.py

```diff
@@ -17,33 +20,43 @@ class SupplierFrameworkData(object):
 
     data = None
 
-    def __init__(self, client, target_framework_slug):
+    def __init__(self, client, target_framework_slug, *, supplier_ids=None):
+        """
+        :param client: Client object for the Digital Marketplace API
+        :param target_framework_slug str: Framework slug
+        :param supplier_ids: List of supplier IDs to filter data by
+        """
         self.client = client
         self.target_framework_slug = target_framework_slug
+        self.supplier_ids = supplier_ids
 
-    def get_supplier_frameworks(self):
+    def get_supplier_frameworks(self, supplier_ids=None):
         """Return supplier frameworks."""
-        return self.client.find_framework_suppliers(
+        framework_suppliers = self.client.find_framework_suppliers(
             self.target_framework_slug, with_declarations=None
         )['supplierFrameworks']
+        if supplier_ids:
+            framework_suppliers = [s for s in framework_suppliers if s["supplierId"] in supplier_ids]
+        return framework_suppliers
 
-    def get_supplier_users(self):
+    def get_supplier_users(self, supplier_ids=None):
```

Comment:
To me it's a _little_ funny that these methods (`get_supplier_users`, `get_supplier_frameworks`) ignore their `self.supplier_ids` and instead expect an argument. It's probably only odd when they're considered from the perspective of being "public" methods though. Others may disagree, though.

Reply:
That's fair, I'll change it.

Reply:
Done